src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,123 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int treshhold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[capacity];
+    }
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V item;
+        private Node<K, V> next;
+
+        public Node(K key, V item, Node<K, V> next) {
+            this.key = key;
+            this.item = item;
+            this.next = next;
+            hash = hashCode();
```

Comment:
it's a bad practice to call methods for the calculation of values inside of the constructor, you should pass the already calculated value here, just to initialize the field of an object